contrib/examples/walkthrough/ups-instance.yaml

```diff
@@ -7,6 +7,7 @@ spec:
   serviceClassName: user-provided-service
   planName: default
   parameters:
-    credentials:
-      name: root
-      password: letmein
+    inline:
+      credentials:
+        name: root
+        password: letmein
```

Comment:
Having the "inline" property in there feels a little clunky. Is it not possible to do something like:
```
spec:
  serviceClassName: user-provided-service
  planName: default
  parameters:
    credentials:
      name: root
      password: letmein
    k8s_secretRef:
      name: mySecret
    k8s_secretKeyRef:
      name: myOtherSecret
      key: key1
```
Basically just have some well-defined property names that we reserve. Note, that if we do this there's no reason not to allow for all 3 variants to appear at the same time.

Also, it seems a little odd that we've said that there may be sensitive data in the binding credentials along side non-sensitive data (so we're asking for there to be a way to identify each in the spec), and yet we're not providing a way for someone to do the same thing on the input side of things - if we adopt a "union only" approach.  We can deal with any potential conflicts by just making it clear what the rules are - e.g. error out on conflict.